homeassistant/components/generic_thermostat/climate.py

```diff
@@ -177,11 +205,25 @@ def __init__(
         self._max_temp = max_temp
         self._target_temp = target_temp
         self._unit = unit
-        self._support_flags = SUPPORT_FLAGS
-        if away_temp:
+
+        if presets and len(presets) > 0:
+            self._presets = presets
+        else:
+            self._presets = collections.OrderedDict()
+            if away_temp:
+                self._presets[PRESET_AWAY] = away_temp
```

Comment:
I do not fully agree. if you simply update HA setup without changing the configuration, it still work as before. 
But if you change your configuration to add `presets` key, you know that the new feature exists, because you have read the docs (https://deploy-preview-15188--home-assistant-docs.netlify.app/integrations/generic_thermostat/), and you know that you can't use both key.